### ISSUE TITLE
Uncraft recipe will yield solder if "using": [ "soldering_standard" ]

### ISSUE THREAD

**Issue by cainiaowu** [CleverRaven/Cataclysm-DDA#25576]
**Describe the bug**  
Uncraft recipe will yield solder if "using" requirements.

**To Reproduce**  
1. Debug spawn a car alternator, a soldering iron with charge and a toolbox.
2. Try disassemble the car alternator.
3. There's solder in the yield list.

**Expected behavior**  
Solder should not in the yield list. 
From its description: "A small spool of solder, able to be used in ammunition and electronics."
Melted solder is hard to recover and even if you did you cant get a nice "spool" one.

**Screenshots**  
car alternator:
![image](https://user-images.githubusercontent.com/4143592/45474035-3e55e380-b76b-11e8-9760-fd1f22f58b66.png)

7.5 kw generator:

![image](https://user-images.githubusercontent.com/4143592/45474103-67767400-b76b-11e8-8f07-69f8ee046b1a.png)

**Versions and configuration(please complete the following information):**
 - OS: Any
 - Game Version: 0.c-7904
 - Graphics version: Any
 - Mods loaded: Default mod list 

**Additional context**  
For reference:
```

{
  "type": "uncraft",
  "result": "alternator_car",
  "skill_used": "mechanics",
  "difficulty": 3,
  "time": 30000,
  "qualities": [
    { "id": "SCREW", "level": 1 },
    { "id": "WRENCH", "level": 1 },
    { "id": "SAW_M", "level": 1 }
  ],
  "using": [
    [ "soldering_standard", 40 ],
    [ "welding_standard", 10 ]
  ],
  "components": [
    [ [ "power_supply", 1 ] ],
    [ [ "cable", 40 ] ],
    [ [ "bearing", 20 ] ],
    [ [ "scrap", 2 ] ],
    [ [ "steel_chunk", 2 ] ]
  ]
}

{
  "type": "uncraft",
  "result": "generator_7500w",
  "skill_used": "mechanics",
  "difficulty": 4,
  "time": 240000,
  "qualities": [
    { "id": "SCREW", "level": 1 },
    { "id": "WRENCH", "level": 1 },
    { "id": "SAW_M", "level": 1 }
  ],
  "tools": [
    [ [ "soldering_iron", 80 ], [ "toolset", 80 ], [ "oxy_torch", 40 ] ]
  ],
  "components": [
    [ [ "power_supply", 4 ] ],
    [ [ "cable", 400 ] ],
    [ [ "bearing", 80 ] ],
    [ [ "frame", 1 ] ],
    [ [ "scrap", 12 ] ],
    [ [ "steel_chunk", 8 ] ]
  ]
}


```

**Comment by FulcrumA** [comment]
I am unsure. Retrieving at least some of the solder is not an uncommon practice. There's even small handpumps designed just for that purpose when soldering.
Also, it seems you've left some of the template needlessly.

**Comment by kevingranade** [comment]
Solder extracted with a suction device is normally discarded, trying to re-use adulterated solder is asking for trouble.

**Comment by Mecares** [comment]
Looks like it was intended to consume solder to uncraft not yield it.

**Comment by FulcrumA** [comment]
That doesn't seem sensible either. You wouldn't need to solder anything to take it apart. Soldering is for connecting various bits.

**Comment by kevingranade** [comment]
You use a soldering iron to disassemble electronics too.  Re-melt the
solder and then wick or suck it away.

**Comment by FulcrumA** [comment]
Yes, you use the soldering iron tool, but you don't use additional solder to disassemble stuff.

**Comment by l29ah** [comment]
It would make sense to recover about a half of the spent solder, since you can't easily get it off the copper pads/leads into reusable blobs when it covers the pads as a thin film.

**Comment by Theundyingcode** [comment]
> It would make sense to recover about a half of the spent solder, since you can't easily get it off the copper pads/leads into reusable blobs when it covers the pads as a thin film.

You can "recover" it but it would not really be useable.  You would get a aquard blob instead of wire, sheets or any other usable shape.  Plus you are just asking for added impurities and aquard mixes of high and low temperature solder. (different melting points are more common in larger soldering jobs where you would likely use a blow torch and thus have less control preventing you from accidentally messing up your previous solder)

**Comment by Kelenius** [comment]
Sounds like these recipes should return something like "low-quality solder" that could be used in recipes like BBs but not soldering.

**Comment by kevingranade** [comment]
That sounfs like a great solution to me.

**Comment by TealcOneill** [comment]
@GuardianDll Fixed by #65486
